File: app/src/main/java/rssdemo/andras/hu/rssdemo/repository/FeedRepository.java
package rssdemo.andras.hu.rssdemo.repository;

import rssdemo.andras.hu.rssdemo.data.Feed;
import rx.Observable;

public class FeedRepository {

    private FeedConverter feedConverter;

    public FeedRepository(FeedConverter feedConverter) {
        this.feedConverter = feedConverter;
    }

    public Observable<Feed> getFeed(final String url) {
        return Observable.fromCallable(() -> feedConverter.convert(url));
    }

}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/repository/FeedRepository.kt
package rssdemo.andras.hu.rssdemo.repository

import rssdemo.andras.hu.rssdemo.data.Feed
import rx.Observable

class FeedRepository(private val feedConverter: FeedConverter) {

    fun getFeed(url: String): Observable<Feed> {
        return Observable.fromCallable<Feed> { feedConverter.convert(url) }
    }

}
